msal/src/Microsoft.Identity.Client/Platforms/Android/UIParent.cs

```diff
@@ -115,27 +99,16 @@ public UIParent(Activity activity, bool useEmbeddedWebview) : this(activity)
         public static bool IsSystemWebviewAvailable()
         {
             bool isBrowserWithCustomTabSupportAvailable = IsBrowserWithCustomTabSupportAvailable();
-            if (!isBrowserWithCustomTabSupportAvailable && !IsChromeEnabled())
-            {
-                return false;
-            }
-            else if (isBrowserWithCustomTabSupportAvailable)
-            {
-                return true;
-            }
-            else
-            {
-                return false;
-            }
+            return (isBrowserWithCustomTabSupportAvailable || IsChromeEnabled()) &&
```

Comment:
Isn't this the same as "isBrowserWithCustomTabSupportAvailable" by itself since you're AND'ing to it...

Reply:
This is just R# based refactoring ... but I did have a look at this. AFAIK (a ^ b) | a = (a | a) ^ (b | a) = a ^ (b | a)  - am I missing a simplification ?